src/libstd/collections/hash/map.rs

```diff
@@ -2341,14 +2341,14 @@ impl<'a, K, V> OccupiedEntry<'a, K, V> {
     /// use std::collections::hash_map::{Entry, HashMap};
     /// use std::rc::Rc;
     ///
-    /// let mut map: HashMap<Rc<String>, u32> = HashMap::new();
-    /// map.insert(Rc::new("Stringthing".to_string()), 15);
+    /// let mut map: HashMap<Rc<str>, u32> = HashMap::new();
```

Comment:
Why these doc changes?

Reply:
I think we shouldn't promote use of `Rc<String>`, rather show that `Rc<str>` is now feasible and preferable.